src/map.h

```diff
@@ -83,7 +83,15 @@ struct MapEditEvent
 	void setPositionModified(v3s16 pos)
 	{
 		p = pos;
-		modified_blocks.insert(getNodeBlockPos(pos));
+		modified_blocks.push_back(getNodeBlockPos(pos));
+	}
+
+	void setModifiedBlocks(const std::map<v3s16, MapBlock *> blocks)
+	{
+		modified_blocks.clear();
```

Comment:
alternatively `assert(modified_blocks.empty()); // only meant for initialization (once)`